pkg/etag_cache/etag_cache.go

```diff
@@ -0,0 +1,51 @@
+package etag_cache
```

Comment:
I already use `zcache` for caching (example - https://github.com/metacontroller/metacontroller/blob/master/pkg/controller/common/customize/cache.go) with time-based eviction. Do you see any reason why not to use the same approach here ?

Reply:
reimplemented in zcache